firebase_auth_oauth_platform_interface/lib/method_channel_firebase_auth_oauth.dart

```diff
@@ -4,37 +4,59 @@ part of firebase_apple_auth_platform_interface;
 class MethodChannelFirebaseAuthOAuth extends FirebaseAuthOAuth {
   FirebaseApp _app;
 
+  OAuthCredential? _credential;
+
   static const MethodChannel _channel =
       const MethodChannel('me.amryousef.apple.auth/firebase_auth_oauth');
 
   MethodChannelFirebaseAuthOAuth._({FirebaseApp? app})
       : _app = app == null ? Firebase.app() : app,
         super._();
 
+  @override
+  OAuthCredential? get credential => this._credential;
```

Comment:
Might be best to create a separate openSignInFlowForCredentials method 
Instead of adding this nullable field.

Nullable fields like this one is ambiguous. Consumers don’t know when it’s null and when it isn’t so you end up adding documentation for when to use which isn’t a nice developer experience.

Reply:
@amrfarid140 Thank for your feedback!

I agree with you that this is not the best developer experience. I thought that for a while, and this was the best I could do to avoid breaking backward compatibility.

Also, I think that having two methods to operate the exact same way would be "weird", since both would execute the exact same code, leaving the developer wondering the difference between each of them.

I don't understand why the API return a Firebase User in this case, since it can be easily fetched from the FirebaseAuth.instance. It would be more straightforward to return the OAuthCredential in the openSignInFlow method, just like Firebase auth returns Auth result in its signIn methods. But, again, I didn't change that to not break backward compatibility.

If you think we should just create a new method name, I suggest something like "signInOAuth" to be more similar with Firebase names. What do you think?

Reply:
Makes sense to me 👍. Let's go with signInOAuth.